deploy-service/common/src/main/java/com/pinterest/deployservice/db/DBAgentDAOImpl.java

```diff
@@ -264,9 +271,14 @@ public long countDeployingAgentWithHostTag(String envId, String tagName, String
 
     @Override
     public long countFinishedAgentsByDeployWithHostTags(String envId, String deployId, String tagName, List<String> tagValues) throws Exception {
-        String tagValuesStr = QueryUtils.genStringGroupClause(tagValues);
-        Long n = new QueryRunner(dataSource).query(String.format(COUNT_FINISHED_AGENTS_BY_DEPLOY_WITH_HOST_TAGS, tagValuesStr),
-            SingleResultSetHandlerFactory.<Long>newObjectHandler(), envId, envId, deployId, tagName);
+        List<Object> params = new ArrayList<>();
+        params.addAll(Arrays.asList(envId, envId, deployId, tagName));
+        params.addAll(tagValues);
```

Comment:
I saw your usage of `ImmutableList.builder`. Maybe you can apply it to here and elsewhere you are chaging.